### PR TITLE
WIP: Docs: Auto generate page for lint rules

### PR DESCRIPTION
**Pull request by EduardoLopes**
Creates a page based on files inside packages/@romejs/compiler/lint/rules
- The file names are converted to kebab-case and arranged in a table in alphabetical order 
- The link for the rule documentation is generated only if there's a file for that rule inside the src/lint/rules folder 
- The (short) description of the rule is get from the yaml that should be at the top each rule documentation file. If a file have no description, the script will tell.

### CODE REVIEW

File: website/lintRulesListGenerator.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
+const lintRulesFolder = '../packages/@romejs/compiler/lint/rules';
```
Comment (sebmck) [inline website/lintRulesListGenerator.js:1]: Nice! Want to move this to `scripts/lint/update.cjs`? You can avoid having to do a lot of this conversion and file system read yourself.

Comment (EduardoLopes) [reply website/lintRulesListGenerator.js:1]: i wanted to. wasn't confident enough. I'll do it, but i can't do it now

**Comment by EduardoLopes** [comment]
ops

**Comment by EduardoLopes** [comment]
End up no using the file utils, because of [this line](https://github.com/romejs/rome/blob/main/scripts/_utils.cjs#L91). That comment is for a js file and the script i wrote is generating a markdown file.

Didn't wanna change that code myself to make it configurable, i'll leave that for you (if you think it's really necessary), that know more about the internal code.

File: scripts/lint/update.cjs [hunk 2]
```diff
@@ -62,3 +66,92 @@ for (const {ruleName} of defs) {
...
+
+//Used for doc headings
+const categoryAlias = {
+	js: "Javascript",
+	"jsx-a11y": "JSX a11y",
+	react: "React",
+	ts: "TypeScript",
+};
+
+let docTemplateTable = null;
```
Comment (ematipico) [inline scripts/lint/update.cjs:99]: I would suggest to make `docTemplateTable` an array of strings and push strings every time you need.

Comment (EduardoLopes) [reply scripts/lint/update.cjs:99]: I see. But i tend not to use array methods, because they are slower in most cases

**Comment by EduardoLopes** [comment]
@sebmck I did a work around the comment issue. But i'm not sure why Generated files check fail

**Comment by EduardoLopes** [comment]
ohh Generated files check is linting the generated files, that's the error

File: scripts/lint/update.cjs (hunk 2 above)
Comment (ematipico) [reply scripts/lint/update.cjs:95]: Sure but this is not a hot path and it is a code run just for maintainers. But that's OK, it was just a suggestion

**Pull request merged**